modules/imaging_uploader/php/imaging_uploader.class.inc

```diff
@@ -148,27 +148,69 @@ class Imaging_Uploader extends \NDB_Menu_Filter_Form
     function _process($values)
     {
         if ($this->_saveFile($values)) {
+            // MRI archive is now in the mri_upload table. If there are existing
+            // records in table notification_spool for that upload (i.e a re-upload
+            // of a scan that the MRI pipeline failed to process), erase them
+            $db = \Database::singleton();
+            $db->delete(
+                'notification_spool',
+                array('ProcessID' => $this->mri_upload_id)
+            );
```

Comment:
@nicolasbrossard Should't we check first that the previous upload was not successful before deleting all the logs? It seems dangerous deleting entries from that table.

Also, if we delete logs of the first entry, shouldn't we at least keep a record somewhere that the first upload did not go well?

Reply:
I don't see why we're deleting logs at all.. we almost never delete things in LORIS because it's too dangerous.

Reply:
@cmadjar If there are log entries for that upload it can only be because the processing was not successful, otherwise the uploader would not allow the overwrite.

@driusan @cmadjar I could add a new column 'Active' in the notification_spool table that would tell whether a log entry is active or belongs to an older version of an upload that has been overwritten. This could provide a way to store the "history" of a series of uploads. Let me know if that works for you.

Reply:
I'm fine with an "Active" column.

Reply:
I like the idea of an active column too. @nicolasbrossard could you implement it easily? Thanks!

Reply:
'Active' column it is then! Implementation does not sound too complicated.

Reply:
Super! Thanks!!

Reply:
Nice nice nice!